File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0;
        int r = nums.size() - 1;

        int mid = (l + r) / 2;

        while(l < r){
            if(nums[mid] > nums[r]){
                l = mid + 1;
            } else {
                r = mid;
            }
            mid = (l + r) / 2;
        }

        return nums[mid];
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        memo = []

        segment = []

        def permutation():
            if len(segment) == len(nums):
                memo.append(segment.copy())
                return
            for i in nums:
                if i not in segment:
                    segment.append(i)
                    permutation()
                    segment.pop()
        
        permutation()
        return memo